proposals/NNNN-swiftpm-package-traits.md

```diff
@@ -0,0 +1,479 @@
+# Package traits
+
+* Proposal: [SE-NNNN](NNNN-swiftpm-package-traits.md)
+* Authors: [Franz Busch](https://github.com/FranzBusch)
+* Review Manager: TBD
+* Status: **Implemented** https://github.com/apple/swift-package-manager/pull/7392
+
+## Introduction
+
+Over the past years the package ecosystem has grown tremendously in both the
+amount of packages and the functionality that individual packages offer.
+Additionally, Swift is being used in more environments such as embedded systems
+or Wasm. This proposal aims to give package authors a new tool to conditionalize
+the features they offer and the dependencies that they use.
+
+## Motivation
+
+There are various motivating use-cases where package authors might want to use traits to
+express conditional compilation or optional dependencies. This section highlights a few.
```

Comment:
This wording (and some of the paragraphs below) presupposes that readers are aware of what traits/optional features/etc are before having introduced them.